gobblin-runtime/src/main/java/gobblin/scheduler/JobScheduler.java

```diff
@@ -377,11 +373,9 @@ public void runJob(Properties jobProps, JobListener jobListener, JobLauncher job
       return;
     }
 
-    // Populate the assigned job ID
-    jobProps.setProperty(ConfigurationKeys.JOB_ID_KEY, JobLauncherUtils.newJobId(jobName));
-
     // Launch the job
     try (Closer closer = Closer.create()) {
+      closer.register(MDC.putCloseable(ConfigurationKeys.JOB_NAME_KEY, jobName));
```

Comment:
Don't think that is needed here.  It is added as part of the `AbstractJobLauncher.launchJob` call done here: `closer.register(jobLauncher).launchJob(jobListener);`.